kodilla-good-patterns/src/main/java/flightSearcher/FlightProcessor.java

```diff
@@ -1,26 +1,49 @@
 package flightSearcher;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class FlightProcessor {
 
-    CreateFlightList createFlightList;
-    FindFlight findFlight;
+    CreateFlightList createFlightList = new CreateFlightList();
 
-    public FlightProcessor(CreateFlightList createFlightList, FindFlight findFlight) {
-        this.createFlightList = createFlightList;
-        this.findFlight = findFlight;
+    public FlightProcessor() {
     }
 
-    public List<Flight> findFlightFrom(Flight flight) {
-        return findFlight.findFlightFrom(flight.getFrom());
+    List<Flight> inIt() {
+        List<Flight> temporaryList = createFlightList.getAllFlights();
+        return temporaryList;
     }
 
-    public List<Flight> findFlightTo(Flight flight) {
-        return findFlight.findFlightTo(flight.getTo());
+    public List<Flight> findFlightFrom(String flightFrom) {
+        List<Flight> flights = inIt();
+
+        return flights.stream()
+                .filter(flight1 -> flight1.getFrom().equals(flightFrom))
+                .collect(Collectors.toList());
     }
 
-    public List<FlightPair> findIndirect() {
-        return findFlight.findIndirectFlights();
+    public List<Flight> findFlightTo(String flightTo) {
+        List<Flight> flights = inIt();
+
+        return flights.stream()
+                .filter(flight1 -> flight1.getTo().equals(flightTo))
+                .collect(Collectors.toList());
+    }
+
+    public List<FlightPair> findIndirect(Flight flightFromTo) {
+        List<Flight> flights = inIt();
+        List<FlightPair> pairList = new ArrayList<>();
+        for(Flight fl : flights) {
+            flights.stream()
+                    .filter(f -> f.getFrom().equals(flightFromTo.getFrom()))
+                    .filter(fp -> fp.getFrom().equals(flightFromTo.getFrom()) && fp.getTo().equals(fl.getFrom()))
+                    .forEach(f -> pairList.add(new FlightPair(f, flightFromTo)));
+        }
+
+        return pairList.stream()
+                .distinct()
+                .collect(Collectors.toList());
```

Comment:
Tu masz dzialajace dobrze rozwiazanie - przeksztalc to teraz na streamy

    public List<FlightPair> findIter(String from, String to) {
        List<FlightPair> intermediateFlights = new ArrayList<>();
        List<Flight> flightsFrom = findFrom(from);
        for (Flight flightFrom : flightsFrom) {
            intermediateFlights.addAll(findFrom(flightFrom.getArrival()).stream()
                    .filter(flight -> flight.getArrival().equals(to))
                    .map(flight -> new FlightPair(flightFrom, flight))
                    .collect(Collectors.toList()));
        }

        return intermediateFlights;
    }

Reply:
Nie do końca rozumiem, mam pozbyć się pętli?